libsnark/polynomial_commitments/kzg10_batched.hpp

```diff
@@ -0,0 +1,113 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef __LIBSNARK_POLYNOMIAL_COMMITMENTS_KZG10_BATCHED_HPP__
+#define __LIBSNARK_POLYNOMIAL_COMMITMENTS_KZG10_BATCHED_HPP__
+
+#include "libsnark/polynomial_commitments/kzg10.hpp"
+
+/// Reference:
+/// - [GWC19]:
+///   Title: "PLONK: Permutations over Lagrange-bases for Oecumenical
+///           Noninteractive arguments of Knowledge"
+///   eprint: https://eprint.iacr.org/2019/953.pdf
+
+namespace libsnark
+{
+
+/// Batched version of KZG10, as described in Section 3 of [GWC19]. Supports
+/// evaluation of arbitrary number of polynomials at 2 points, requiring a
```

Comment:
The number of polynomials and evaluation points can both be polynomial in the security parameter. Why do you stick to 2 here? Is that to stick to the Plonk use-case?

Reply:
Yes.  The scheme given in Section 3 of [GWC19] uses 2 evaluation points.  It can indeed be extended to handle a dynamic number, but the data structures and code are much more complex, so for now it uses this form.

Reply:
(In the case of more evaluation points, we could later implement the generalization, or it may be better to use the [BDFG21] scheme from #44)